idem_azurerm/exec/azurerm/containerregistry/registry.py

```diff
@@ -0,0 +1,891 @@
+    .. code-block:: bash
+
+        azurerm.containerregistry.registry.list
+
+    """
+    result = {}
+    regconn = await hub.exec.azurerm.utils.get_client(
+        ctx, "containerregistry", **kwargs
+    )
+    try:
+        if resource_group:
+            registries = await hub.exec.azurerm.utils.paged_object_to_list(
+                regconn.registries.list_by_resource_group(
+                    resource_group_name=resource_group
+                )
+            )
+        else:
+            registries = await hub.exec.azurerm.utils.paged_object_to_list(
+                regconn.registries.list()
+            )
+        for registry in registries:
+            result[registry["name"]] = registry
+    except CloudError as exc:
+        await hub.exec.azurerm.utils.log_cloud_error(
+            "containerregistry", str(exc), **kwargs
+        )
+        result = {"error": str(exc)}
+
+    return result
+
+
+async def list_credentials(hub, ctx, name, resource_group, **kwargs):
+    """
+    .. versionadded:: 3.0.0
+
+    Lists the login credentials for the specified container registry.
+
+    :param name: The name of the container registry.
+
+    :param resource_group: The name of the resource group to which the container registry belongs.
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        azurerm.containerregistry.registry.list_credentials testrepo testgroup
+
+    """
+    result = {}
+    regconn = await hub.exec.azurerm.utils.get_client(
+        ctx, "containerregistry", **kwargs
+    )
+    try:
+        creds = regconn.registries.list_credentials(
+            registry_name=name, resource_group_name=resource_group
+        )
+        result = creds.as_dict()
+    except CloudError as exc:
+        await hub.exec.azurerm.utils.log_cloud_error(
+            "containerregistry", str(exc), **kwargs
+        )
+        result = {"error": str(exc)}
+
+    return result
+
+
+async def list_usages(hub, ctx, name, resource_group, **kwargs):
+    """
+    .. versionadded:: 3.0.0
+
+    Lists the quota usages for the specified container registry.
+
+    :param name: The name of the container registry.
+
+    :param resource_group: The name of the resource group to which the container registry belongs.
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        azurerm.containerregistry.registry.list_usages testrepo testgroup
+
+    """
+    result = {}
+    regconn = await hub.exec.azurerm.utils.get_client(
+        ctx, "containerregistry", **kwargs
+    )
+    try:
+        usages = regconn.registries.list_usages(
+            registry_name=name, resource_group_name=resource_group
+        )
+        result = usages.as_dict()
+    except CloudError as exc:
+        await hub.exec.azurerm.utils.log_cloud_error(
+            "containerregistry", str(exc), **kwargs
+        )
+        result = {"error": str(exc)}
+
+    return result
+
+
+async def regenerate_credential(
+    hub, ctx, name, resource_group, credential="password", **kwargs
```

Comment:
Is there a reason you decided to default this instead of making it a required parameter for the user to decide?

Reply:
there isn't much harm in a sane default